src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-daemon-list/service-daemon-list.component.html

```diff
@@ -15,3 +54,27 @@
     {{ row.status_desc }}
   </span>
 </ng-template>
+
+<ng-template #listTpl
+             let-value="value">
+  <div *ngIf="value.length == 0 || value == undefined">
+    <span>No data available</span>
+  </div>
+  <div *ngIf="value.length != 0 && value != undefined">
+    <div *ngFor="let event of value">
+      <b>{{ event.created | relativeDate }} - </b>
+      <span class="badge badge-info">{{ event.subject }}</span><br>
+      <div *ngIf="event.level == 'INFO'"
+           class="indent">
+      <i class="fa fa-info-circle"
+         aria-hidden="true"></i>
+      </div>
+      <div *ngIf="event.level == 'ERROR'"
+           class="indent">
+      <i class="fa fa-exclamation-triangle"
+         aria-hidden="true"></i>
+      </div>
+      <div class="indent"><p>{{ event.message }}</p></div>
+    </div>
```

Comment:
Don't we prefer to use ul > li here? That we you wouldn't need the upper wrapping ngIf (I think ngFor handles that), and by using a list here, the indentantion style can be achieved naturally, by ul > li > ul > li ([bootstrap](https://getbootstrap.com/docs/4.0/components/list-group/), no equivalent found in Patternfly).

Reply:
Added the ul>li approach now. But found out that the wrapping ngIf still needs to be there.